### ISSUE TITLE
Example of RL on MATH dataset not reproducable.

### ISSUE THREAD

**Issue by HaoquanZhang** [thinking-machines-lab/tinker-cookbook#180]
`python -m tinker_cookbook.recipes.math_rl.train env=math model_name="Qwen/Qwen3-8B" group_size=16 groups_per_batch=64 learning_rate=2e-5 max_tokens=512`

result of step 24

`
{"step": 24, ... "env/all/reward/total": -0.0814453125, "env/all/format": 0.0, }
`

**Comment by bledden** [comment]
@HaoquanZhang put up a fix for this after reproducing the non-reproducible example - https://github.com/thinking-machines-lab/tinker-cookbook/pull/203
